test/inductor/test_codecache.py

```diff
@@ -37,3 +49,209 @@ def test_codecache_spawn():
 @requires_cuda()
 def test_codecache_fork():
     _run_codecache_test("fork")
+
+
+class TestFxGraphCache(TestCase):
+    @requires_cuda()
+    @config.patch({"fx_graph_cache": True})
+    @patch("torch._inductor.codecache.cache_dir")
+    def test_cache_load(self, mock_cache_dir):
+        """
+        Verify that we can populate and load CompiledFxGraphs from the cache.
+        """
+
+        def fn(x, y):
```

Comment:
Would love input on whether people think we should test compiling/caching more sophisticated code. Also, is it valuable to run an instance of this test with the tensors on cpu? Different dtypes, perhaps?

Reply:
Yeah I would definitely set up tests that have inputs with 1. different strides, 2. different dtypes, 3. different storage offsets, 4. different aliasing of inputs.

Reply:
I think we should add an option in the runner that would compile the model, torch._dynamo.reset(), then check that when run again we both hit the cache and accuracy passes. And we should enable that wherever we're checking accuracy right now. 

Any other thoughts on testing @ezyang ?

Reply:
@masneral @eellison my original tests were supposed to be doing just this, compiling, reset, compiling again. This is where I hit all those cudagraph errors.

Reply:
I'll add `torch._dynamo.reset()` to the test(s), but I think @eellison is also proposing to add a new feature to the benchmark runner?

Reply:
Right. Would be great to test this on models in ci.

Reply:
NGL, I don't really think any of our existing CI is going to do all that well at catching errors in caching. There are two things you want to test:

1. Cache hits
2. Cache missing when you should

But existing CI isn't going to have a pre-warmed cache to test (1) nor (2), because in CI you mostly want to run everything once, cold.

To test cache hits, I suppose you could add a testing mode where instead of directly running a compiled product, you serialize it out, deserialize it, and check the deserialization still passes tests. But this won't catch problems that occur only when you change process. And running every test twice... well... this doesn't seem like good ROI, except maybe some occasional fuzz test (not even sure if I'd want it master only).

To test cache misses, I suppose you can fuzz this by chucking every compiled product into the cache and checking that hits give the same result. This will be a huge pain to debug if it does fail though (because you need to find exactly the right other test which produced the incompatible product) and I'd imagine most of the time things were different enough that you didn't really hit any nasty edge cases.

I think a more promising approach is to assume that we *are* going to do it wrong, and figure out how to make it as pleasant and easy to debug when we have done it wrong. Stuff like recording provenance where cached products came from, logging when you use a cached product, an easy way to ignore cache, etc.